server/fleet/queries.go

```diff
@@ -357,3 +369,64 @@ type QueryStats struct {
 	UserTime     int       `json:"user_time" db:"user_time"`
 	WallTime     int       `json:"wall_time" db:"wall_time"`
 }
+
+// MapQueryReportsResultsToRows converts the scheduled query results as stored in Fleet's database
+// to HostQueryResultRows to be exposed to the API.
+func MapQueryReportResultsToRows(rows []*ScheduledQueryResultRow) ([]HostQueryResultRow, error) {
+	var results []HostQueryResultRow
+	for _, row := range rows {
+		var columns map[string]string
+		if err := json.Unmarshal(row.Data, &columns); err != nil {
+			return nil, err
+		}
+		results = append(results, HostQueryResultRow{
+			HostID:      row.HostID,
+			Hostname:    row.Hostname,
+			LastFetched: row.LastFetched,
+			Columns:     columns,
+		})
+	}
+	return results, nil
+}
+
+// HostQueryResultRow contains a single scheduled query result row from a host.
+// This type is used to expose the results on the API.
+type HostQueryResultRow struct {
+	// HostID is the unique ID of the host.
+	HostID uint `json:"host_id"`
+	// Hostname is the host's hostname.
+	Hostname string `json:"host_name"`
+	// LastFetched is the time this result row was received.
+	LastFetched time.Time `json:"last_fetched"`
+	// Columns contains the key-value pairs of a result row.
+	// The map key is the name of the column, and the map value is the value.
+	Columns map[string]string `json:"columns"`
+}
+
+// ScheduledQueryResult holds results of a scheduled query received from a osquery agent.
+type ScheduledQueryResult struct {
+	// QueryName is the name of the query.
+	QueryName string `json:"name,omitempty"`
+	// OsqueryHostID is the identifier of the host.
+	OsqueryHostID string `json:"hostIdentifier"`
+	// Snapshot holds the result rows. It's an array of maps, where the map keys
+	// are column names and map values are the values.
+	Snapshot []json.RawMessage `json:"snapshot"`
+	// LastFetched is the time this result was received.
+	LastFetched uint `json:"unixTime"`
+}
+
+// ScheduledQueryResultRow is a scheduled query result row.
+type ScheduledQueryResultRow struct {
+	// QueryID is the unique identifier of the query.
+	QueryID uint `db:"query_id"`
+	// HostID is the unique identifier of the host.
+	HostID uint `db:"host_id"`
+	// Hostname is the host's hostname.
+	Hostname string `db:"hostname"`
+	// Data holds a single result row. It holds a map where the map keys
+	// are column names and map values are the values.
+	Data json.RawMessage `db:"data"`
```

Comment:
Reading this i'm realizing maybe I should have named the "Data" column in `query_results` to "Columns"